### PR TITLE
[RFR] Update migrations to support typed models

### PR DESCRIPTION
**Pull request by cwisecarver**
This makes the children of Node be separate classes that put their data in the Node table. It adjusts the migrations to support this.

While waiting for migrations to run I've been translating other models, so there are some of those in here too.

It also adds a BSONGuidMixin to add the fields and behavior that things with a BSONGuid need.

### CODE REVIEW

**Comment by cwisecarver** [comment]
@sloria The migrations are working again after switching to typed models. I also added a few more models in here without migrations for them, but they mostly (if not all) inherit from BaseModel which now has a classmethod called migrate_from_modm that will take a modm object and return an unsaved django model instance with all the fields except relations filled out. This should make migration logic more contained while still allowing easy overriding of migration behavior. lmk if you think it's okay to merge. I rebased it onto master into it a few minutes ago.

File: osf_models/management/commands/migrate_bare_objects.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from website.app import init_app
+from osf_models.scripts.load_guids import main as load_guids
+from osf_models.scripts.load_blacklist_guids import \
+    main as load_blacklist_guids
+
+class Command(BaseCommand):
+    help = 'Migrates bare objects from tokumx to postgres'
+
+    def handle(self, *args, **options):
+        print 'Initializing Flask App...'
```
Comment (sloria) [inline osf_models/management/commands/migrate_bare_objects.py]: Minor: Let's use the `print()` function moving forward, for easier Py3 compat in the future.

File: osf_models/models/base.py [hunk 2]
```diff
@@ -137,3 +125,64 @@ def save(self, *args, **kwargs):
...
+
+    @classmethod
+    def migrate_from_modm(cls, modm_obj):
+        """
+        Given a modm object, make a django object with the same local fields.
+        This is a base method that may work for simple things. It should be customized for complex ones.
+        :param modm_obj:
+        :return:
+        """
+        guid, created = Guid.objects.get_or_create(guid=modm_obj._id)
```
Comment (sloria) [inline osf_models/models/base.py]: Shouldn't this only be done for GUID models? Maybe add this to `GuidMixin`?

**Comment by sloria** [comment]
Looks good on first glance. Main concern is https://github.com/CenterForOpenScience/osf-models/pull/59/files#r73178115 . Once that is addressed and the build is passing, go ahead and merge.

**Comment by sloria** [comment]
Also, might be good to add test(s) for `migrate_from_modm` since it's such a critical method. That can go in a follow-up PR or commit if you want, though.

File: osf_models/models/base.py (hunk 2 above)
Comment (cwisecarver) [inline osf_models/models/base.py]: Done.

File: osf_models/management/commands/migrate_bare_objects.py (hunk 1 above)
Comment (cwisecarver) [inline osf_models/management/commands/migrate_bare_objects.py]: I'm 99% sure I got them all, based on `ag "print\s[\'\"]"`

**Pull request merged**